cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/RangeAnalysisStage.qll

```diff
@@ -1020,6 +1020,13 @@ module RangeStage<DeltaSig D, BoundSig<D> Bounds, LangSig<D> LangParam, UtilSig<
         or
         upper = false and delta = D::fromFloat(D::toFloat(d1).minimum(D::toFloat(d2)))
       )
+      or
+      exists(SemExpr mid, D::Delta d |
+        e.(SemNegateExpr).getOperand() = mid and
+        b instanceof SemZeroBound and
+        bounded(mid, b, d, upper.booleanNot(), fromBackEdge, origdelta, reason) and
+        delta = D::fromFloat(-D::toFloat(d))
+      )
     )
```

Comment:
Let's see if I get this. If `e` is unary minus and `b` is known to be zero, and the delta before applying the unary minus was `d`, then the delta after is `-d`.

Reply:
Yep, that's my understanding as well 😄